### PR TITLE
doc/dev/crimson: add a section for testing with crimson-store-nbd

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/crimson/crimson.rst [hunk 1]
```diff
@@ -176,6 +176,68 @@ see `Prometheus`_ for more details.
 Profiling Crimson
 =================
 
+fio
+---
+
+``crimson-store-nbd`` exposes crimson object store internals as an NBD server
```
Comment (rzarzynski) [inline doc/dev/crimson/crimson.rst:182]: Is the `internals` word is necessary? Maybe *crimson object store* or even *crimson's `FuturizedStore` as an NBD server* would be fine?

**Review by rzarzynski — APPROVED** [review]

File: doc/dev/crimson/crimson.rst (hunk 1 above)
Comment (tchaikov) [reply doc/dev/crimson/crimson.rst]: @rzarzynski i changed it to "SeaStore" in hope that it is more specific. IMHO, `FuturizedStore` is an interface which can imply two implementations, but crimson-store-nbd only exercises SeaStore.

File: doc/dev/crimson/crimson.rst [hunk 2]
```diff
@@ -176,6 +176,68 @@ see `Prometheus`_ for more details.
 Profiling Crimson
 =================
 
+fio
+---
+
+``crimson-store-nbd`` exposes SeaStore as an NBD server for use with fio in
+basic benchmarking.
```
Comment (athanatos) [inline doc/dev/crimson/crimson.rst:183]: It actually can expose alienstore with --type = "bluestore".  See fs_driver.cc, config.get_fs_type().  How about "exposes configurable FuturizedStore internals as an NBD server for use with fio."

File: doc/dev/crimson/crimson.rst [hunk 3]
```diff
@@ -176,6 +176,68 @@ see `Prometheus`_ for more details.
...
+   the path to the raw device, like ``/dev/nvme1n1`` in place of the created
+   file for testing with a block device.
+
+   .. prompt:: bash $
+
+      export disk_img=/tmp/disk.img
+      export unix_socket=/tmp/store_nbd_socket.sock
+      rm -f $disk_img $unix_socket
+      truncate -s 512M $disk_img
+      ./bin/crimson-store-nbd --device-path $disk_img -c 1 --mkfs true --uds-path ${unix_socket} &
```
Comment (athanatos) [inline doc/dev/crimson/crimson.rst:214]: Best to specify a type here and explain what it does.

File: doc/dev/crimson/crimson.rst (hunk 1 above)
Comment (tchaikov) [reply doc/dev/crimson/crimson.rst]: @rzarzynski sorry, i was wrong. updated and repushed.

File: doc/dev/crimson/crimson.rst (hunk 2 above)
Comment (tchaikov) [reply doc/dev/crimson/crimson.rst]: updated.

File: doc/dev/crimson/crimson.rst (hunk 3 above)
Comment (tchaikov) [reply doc/dev/crimson/crimson.rst]: updated and repushed.

**Comment by tchaikov** [comment]
@athanatos could you take another look?

**Review by athanatos — APPROVED** [review]

**Pull request merged**